lib/ExecutionEngine/Context.cpp

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "glow/ExecutionEngine/Context.h"
+
+using namespace glow;
+
+Tensor *Context::get(Placeholder *P) {
+  auto it = map_.find(P);
+  if (it == map_.end()) {
+    return nullptr;
+  }
+
+  return it->second;
+}
+
+void Context::insert(Placeholder *P, Tensor &&T) {
+  assert(!map_.count(P) && "Placeholder already registered");
+  // Take ownership over the tensor.
+  map_[P] = new Tensor(std::move(T));
+}
+
+bool Context::count(Placeholder *P) { return map_.count(P); }
```

Comment:
This feels a little weird, since std::unordered_map::count() returns a size_type rather than a bool.